gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/test/java/io/gravitee/rest/api/management/rest/resource/auth/ExternalAuthenticationResourceTest.java

```diff
@@ -0,0 +1,426 @@
+import java.io.IOException;
+import java.io.StringWriter;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.SecureRandom;
+import java.security.interfaces.ECPrivateKey;
+import java.security.interfaces.ECPublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Base64;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.bouncycastle.util.io.pem.PemObject;
+import org.bouncycastle.util.io.pem.PemWriter;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.server.model.Resource;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * @author Kamiel Ahmadpour (kamiel.ahmadpour at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class ExternalAuthenticationResourceTest extends AbstractResourceTest {
+
+    private static final String ORGANIZATION_ID = "organization-id";
+    private static final String ENVIRONMENT_ID = "environment-id";
+    private static final String JWT_ISSUER = "test-issuer";
+    private static final String JWT_SUBJECT = "test-user-id";
+    private static final String EC_KID = "ec-256";
+    private static final String RSA_KID = "rsa-256";
+    private static final String PATH = "auth/external";
+    private static final Map<String, KeyPair> KEY_PAIRS = new ConcurrentHashMap<>();
+
+    @Autowired
+    private InstallationAccessQueryService installationAccessQueryService;
+
+    @Override
+    protected String contextPath() {
+        return "";
+    }
+
+    @Override
+    protected void decorate(ResourceConfig resourceConfig) {
+        super.decorate(resourceConfig);
+        Resource.Builder resourceBuilder = Resource.builder(JWKSResource.class);
+        resourceConfig.registerResources(resourceBuilder.build());
+    }
+
+    @BeforeClass
+    public static void initKeyPairs() {
+        try {
+            KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("EC");
+            SecureRandom random = SecureRandom.getInstance("SHA1PRNG");
+            keyPairGenerator.initialize(256, random);
+            KEY_PAIRS.put(EC_KID, keyPairGenerator.generateKeyPair());
+
+            keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+            keyPairGenerator.initialize(2048);
+            KEY_PAIRS.put(RSA_KID, keyPairGenerator.generateKeyPair());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Before
+    public void init() {
+        System.setProperty("auth.external.enabled", "true");
+        System.setProperty("auth.external.issuer", JWT_ISSUER);
+        when(installationAccessQueryService.getConsoleUrl(ORGANIZATION_ID))
+            .thenReturn(rootTarget(".well-known/jwks.json").getUriBuilder().toTemplate());
```

Comment:
I don't understand this mock. Shouldn't be a simple HTTP URL instead of something extracted from jwks.json ?

Reply:
No because the port is not fix and it will change with any new run. so there is no fix url in fact